library/systemd/src/lib/yast2/system_service.rb

```diff
@@ -274,30 +319,82 @@ def active=(value)
     # Sets start mode to the underlying system
     def save_start_mode
       return unless changes[:start_mode]
+
       result =
         case changes[:start_mode]
         when :on_boot
-          service.enable && socket.disable
+          service.enable && (socket ? socket.disable : true)
         when :on_demand
-          service.disable && socket.enable
+          service.disable && (socket ? socket.enable : true)
         when :manual
-          service.disable && socket.disable
+          service.disable && (socket ? socket.disable : true)
         end
+
       register_error(:start_mode) unless result
     end
 
-    # Sets service status
-    def set_current_status
-      return if changes[:active].nil?
-      result =
-        if changes[:active] && !service.active?
-          service.start
-        elsif changes[:active] == false && service.active?
-          service.stop
-        end
+    # Performs the indicated action (if any) in the underlying system
+    #
+    # @note In case the action cannot be performed, an error is registered,
+    #   see {#register_error}.
+    def perform_action
+      return unless action
+
+      result = send("perform_#{action}")
+
       register_error(:active) if result == false
     end
 
+    # Starts the service in the underlying system
+    #
+    # @return [Boolean] true if the service was correctly started
+    def perform_start
+      result = true
+
+      if socket && start_mode == :on_demand
+        result &= socket.start unless socket_active?
```

Comment:
I think bitwise and is wrong here and should be `&&=` logical and instead

Reply:
I didn't realize I was doing bitwise operations. Thanks!